src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -220,8 +231,16 @@ struct task_metrics {
     this->num_times_split_retry_throw += other.num_times_split_retry_throw;
     this->time_blocked_nanos += other.time_blocked_nanos;
     this->time_lost_nanos += other.time_lost_nanos;
+    this->time_lost_or_blocked += other.time_lost_or_blocked;
     this->gpu_max_memory_allocated =
       std::max(this->gpu_max_memory_allocated, other.gpu_max_memory_allocated);
+    // each task_metric represents a separate thread that contributed to a task
+    // We don't know if those threads were run at the same time or not so
+    // to be conservative in the measurement we are adding them. If we assumed
```

Comment:
with that said, should we also change gpu_max_memory_allocated to use sum instead of max?